src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -1575,15 +1581,13 @@ public static NamedLambda[] Parse(FormulaValue[] args)
         }
     }
 
-#pragma warning disable CS0618 // Type or member is obsolete
-    internal class JoinImpl : JoinFunction, IAsyncTexlFunctionJoin
-#pragma warning restore CS0618 // Type or member is obsolete
+    internal class JoinImpl : JoinFunction, IFunctionInvoker
     {
         public override Type DeclarationType => typeof(JoinFunction);
 
-        public async Task<FormulaValue> InvokeAsync(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        public async Task<FormulaValue> InvokeAsync(FunctionInvokeInfo invokeInfo, CancellationToken cancellationToken)

```

Comment:
>cancellationToken [](http://example.com/codeflow?start=101&length=17)

this is unused - should be tested & passed along

Reply:
good catch. 
Existing code. JoinTables is checking cancellation via EvalVisitor.CheckCancel.